libraries/apollo-ast/src/commonMain/kotlin/com/apollographql/apollo3/ast/internal/semanticEquals.kt

```diff
@@ -21,7 +22,13 @@ import com.apollographql.apollo3.ast.GQLStringValue
 import com.apollographql.apollo3.ast.GQLVariableValue
 import com.apollographql.apollo3.ast.toUtf8
 
-internal fun GQLNode.semanticEquals(other: GQLNode): Boolean {
+/**
+ * Returns true if the two nodes are semantically equal, which ignores the source location and the description.
+ * Note that not all cases are implemented - currently [GQLEnumTypeDefinition] and [GQLDirectiveDefinition] are fully supported, and
```

Comment:
Feels like we should support `GQLInputTypeDefinition` and `GQLScalarTypeDefinition` as well given they can be directive arguments.